tests/convoluted_example.rs

```diff
@@ -0,0 +1,95 @@
+mod example {
+    use std::marker::PhantomData;
+    use std::mem::MaybeUninit;
+
+    use state_shift::{states, switch_to, type_state};
+
+    pub struct MyParentObject<'base> {
+        inner: MaybeUninit<*mut core::ffi::c_void>,
+        _marker: PhantomData<&'base ()>,
+    }
+
+    impl<'base> MyParentObject<'base> {
+        pub fn new() -> Self {
+            MyParentObject {
+                inner: MaybeUninit::zeroed(),
+                _marker: PhantomData,
+            }
+        }
+
+        // ...
+
+        pub fn method(&'base self) -> MethodBuilder {
+            MethodBuilder::new()
+        }
+    }
+
+    struct Method {}
+
+    impl Method {
+        fn start(self) -> Result<(), String> {
```

Comment:
I think this should have been `pub fn start`

the rest of the errors we get are due to this library, the fix should be fairly easy. Will try to implement now